### PR TITLE
fix: raise an exception when parsing source encounters a JSONDecode TDE-448

### PR DESCRIPTION
**Pull request by paulfouquet**

### CODE REVIEW

File: scripts/standardising.py [hunk 1]
```diff
@@ -59,7 +59,11 @@ def main() -> None:
     parser.add_argument("--source", dest="source", nargs="+", required=True)
     arguments = parser.parse_args()
 
-    source = format_source(arguments.source)
+    try:
+        source = format_source(arguments.source)
+    except Exception as e:
```
Comment (blacha) [inline scripts/standardising.py:64]: what happens if we dont catch it here? its already logged, so the CLI will just die right? what is the desired behaviour?

Comment (paulfouquet) [reply scripts/standardising.py:64]: Probably not throwing an exception to the user :)

File: scripts/non_visual_qa.py [hunk 2]
```diff
@@ -152,7 +152,7 @@ def non_visual_qa(files: List[str]) -> None:
     get_log().info("non_visual_qa_end", source=files, duration=time_in_ms() - start_time)
 
 
-def main() -> None:
+def main() -> None:  # pylint: disable=duplicate-code
```
Comment (paulfouquet) [inline scripts/non_visual_qa.py:155]: That's annoying that pylint does not raise this warning locally via `pre-commit` script

File: scripts/standardising.py (hunk 1 above)
Comment (blacha) [reply scripts/standardising.py]: I think a bunch of that main function could throw straight back to the client.

I would generally have something like

```python

try:
  main()
except Exception as e:
  log_exception(e)
  exit(1)
```

where your logging all exceptions, and then exiting, how does `argparse` suggest you do error handling in its CLIs?

Comment (paulfouquet) [reply scripts/standardising.py]: > how does argparse suggest you do error handling in its CLIs?

[argparse Invalid arguments](https://docs.python.org/3/library/argparse.html#invalid-arguments)
Looks like they do something similar?

**Comment by amfage** [comment]
This is now covered in https://github.com/linz/topo-imagery/pull/582

**Pull request closed without merging**